File: generate-ci/sample/typescript/src/calculator.ts
/**
 * Calculator module providing basic arithmetic operations.
 */

/**
 * Adds two numbers together.
 * @param a - The first operand
 * @param b - The second operand
 * @returns The sum of a and b
 */
export function add(a: number, b: number): number {
  return a + b;
}

/**
 * Subtracts the second number from the first.
 * @param a - The minuend
 * @param b - The subtrahend
 * @returns The difference of a and b
 */
export function subtract(a: number, b: number): number {
  return a - b;
}

/**
 * Multiplies two numbers together.
 * @param a - The first factor
 * @param b - The second factor
 * @returns The product of a and b
 */
export function multiply(a: number, b: number): number {
  return a * b;
}

/**
 * Divides the first number by the second.
 * @param a - The dividend
 * @param b - The divisor (must not be zero)
 * @returns The quotient of a divided by b
 * @throws {Error} When dividing by zero
 */
export function divide(a: number, b: number): number {
  if (b === 0) {
    throw new Error('Division by zero is not allowed');
  }
  return a / b;
}

File: generate-ci/sample/python/src/app/calculator.py
"""Simple calculator module."""


def add(a: float, b: float) -> float:
    """Add two numbers and return the result.

    Args:
        a: First number.
        b: Second number.

    Returns:
        The sum of a and b.
    """
    return a + b


def subtract(a: float, b: float) -> float:
    """Subtract b from a and return the result.

    Args:
        a: First number.
        b: Number to subtract.

    Returns:
        The difference of a and b.
    """
    return a - b


def multiply(a: float, b: float) -> float:
    """Multiply two numbers and return the result.

    Args:
        a: First number.
        b: Second number.

    Returns:
        The product of a and b.
    """
    return a * b


def divide(a: float, b: float) -> float:
    """Divide a by b and return the result.

    Args:
        a: Dividend.
        b: Divisor.

    Returns:
        The quotient of a divided by b.

    Raises:
        ValueError: If b is zero.
    """
    if b == 0:
        raise ValueError("Cannot divide by zero")
    return a / b
